src/actions.ts

```diff
@@ -1,21 +1,28 @@
-import { fetchOPDSData } from './fetchData';
+import { fetchOPDSData } from "./fetchData";
 
-export const REQUEST_COLLECTION = 'REQUEST_COLLECTION';
-export const LOAD_COLLECTION = 'LOAD_COLLECTION';
+export const FETCH_COLLECTION_REQUEST = "FETCH_COLLECTION_REQUEST";
+export const FETCH_COLLECTION_SUCCESS = "FETCH_COLLECTION_SUCCESS";
+export const LOAD_COLLECTION = "LOAD_COLLECTION";
 
 export function fetchCollection(url: string) {
   return function(dispatch) {
-    dispatch(requestCollection(url));
-    console.log("fetching url: ", url)
-
-    return fetchOPDSData(url, (data) => {
-      dispatch(loadCollection(data, url));
+    dispatch(fetchCollectionRequest(url));
+    return new Promise((resolve, reject) => {
+      fetchOPDSData(url).then((data: CollectionProps) => {
+        dispatch(loadCollection(data, url));
+        dispatch(fetchCollectionSuccess());
+        resolve(data);
+      }).catch(err => reject(err));
```

Comment:
Maybe this should dispatch a fetch collection failure action instead of rejecting the promise. I'm not sure anything will handle the rejection further up...